src/syscheckd/registry/registry.c

```diff
@@ -756,23 +726,178 @@ void fim_read_values(HKEY key_handle,
+        new.registry_entry.value->scanned = 0;
+        new.type = FIM_TYPE_REGISTRY; //arreglado uninitialized read
+
+        value_path_length = strlen(new.registry_entry.value->path) + strlen(new.registry_entry.value->name) + 2;
+
+        os_malloc(value_path_length, value_path);
+        snprintf(value_path, value_path_length, "%s\\%s", new.registry_entry.value->path, new.registry_entry.value->name);
 
-        fim_registry_process_value_event(new, saved, mode, data_buffer);
+        if (fim_registry_validate_ignore(value_path, configuration, 0)) {
+            os_free(value_path);
+            return;
+        }
+        os_free(value_path);
+
+        if (fim_check_restrict(new.registry_entry.value->name, configuration->restrict_value)) {
+            return;
+        }
+
+        fim_registry_calculate_hashes(&new, configuration, data_buffer);
+
+        fim_registry_get_checksum_value(new.registry_entry.value);
+
+        if (configuration->opts & CHECK_SEECHANGES) {
+            diff = fim_registry_value_diff(new.registry_entry.value->path, new.registry_entry.value->name,
+                                       (char *)data_buffer, new.registry_entry.value->type, configuration);
+        }
+        txn_ctx_regval->diff = diff;
+        txn_ctx_regval->type = data_type;
+
+        int result_transaction = fim_db_transaction_sync_row(regval_txn_handler, &new);
+        if (result_transaction < 0) {
+            mdebug2("dbsync transaction failed due to %d", result_transaction);
+        }
     }
 
     os_free(value_buffer);
     os_free(data_buffer);
 }
 
+cJSON* fim_dbsync_registry_key_json_event(const char* path,
+                                          int arch,
+                                          const cJSON* dbsync_event,
+                                          const registry_t* configuration,
+                                          const event_data_t* evt_data)
+{
+    //cJSON* changed_attributes = NULL;
+
+    cJSON* json_event = cJSON_CreateObject();
+    cJSON_AddStringToObject(json_event, "type", "event");
+
+    cJSON* data = cJSON_CreateObject();
+    cJSON_AddItemToObject(json_event, "data", data);
+
+    cJSON_AddStringToObject(data, "path", path);
+    cJSON_AddStringToObject(data, "arch", (arch == ARCH_32BIT) ? "[x32]" : "[x64]");
+    cJSON_AddNumberToObject(data, "version", 2.0);
+    cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[evt_data->mode]);
+    cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[evt_data->type]);
+
+    //cJSON_AddStringToObject(data, "timestamp",  cJSON_GetObjectItem(fim_entry, "last_event")->valuestring);
+
+    //cJSON_AddItemToObject(data, "attributes", fim_registry_key_attributes_json(new_data, configuration));
+
+    //cJSON_AddItemToObject(data, "changed_attributes", changed_attributes);
+
+    // if (old_data) {
+    //     cJSON_AddItemToObject(data, "old_attributes", fim_attributes_json(old_data));
+    // }
```

Comment:
Are these statements are necessary? If this are not necessary, you could delete them

Reply:
I left them as placeholders because those are needed event fields that cannot be added currently because dbsync library do not provide them right now.